ntoskrnl/ps/query.c

```diff
@@ -566,7 +566,7 @@ NtQueryInformationProcess(IN HANDLE ProcessHandle,
         /* DOS Device Map */
         case ProcessDeviceMap:
 
-            if (ProcessInformationLength != sizeof(PROCESS_DEVICEMAP_INFORMATION))
+            if (ProcessInformationLength != sizeof(((PPROCESS_DEVICEMAP_INFORMATION)0)->Query))
```

Comment:
We have the macro `RTL_FIELD_SIZE` that does just that:
`RTL_FIELD_SIZE(PROCESS_DEVICEMAP_INFORMATION, Query)`

Reply:
Good idea.